### PR TITLE
Use new color scheme in GAPID profiler.

### PR DESCRIPTION
**Pull request by stellama0208**

### CODE REVIEW

File: gapic/src/main/com/google/gapid/perfetto/models/SliceTrack.java [hunk 1]
```diff
@@ -486,4 +488,17 @@ public Node build(Map<Long, List<Builder>> byParent) {
...
+    return colorForSlice(title, depth, 0);
+  }
+
+  public static RGBA getBorderColor(String title, int depth) {
+    return colorForSlice(title, depth, StyleConstants.isLight()? -5 : 5);
+  }
+
+  private static RGBA colorForSlice(String title, int depth, int shadeIdx) {
+    return StyleConstants.Palette.getColor(
+        (title.hashCode() ^ Integer.hashCode(depth)) & 0x7fffffff, shadeIdx);
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/perfetto/models/SliceTrack.java]: `Integer.hashCode` is the identity function (does nothing).

File: gapic/src/main/com/google/gapid/perfetto/models/ThreadInfo.java [hunk 2]
```diff
@@ -96,20 +98,30 @@ public String getDisplay() {
...
 
     if (selected == null) {
-      return baseColor;
+      return colorForThread(queried, 0);
     } else if (queried.utid == selected.utid) {
-      return baseColor;
+      return colorForThread(queried, 0);
     } else if (queried.upid == selected.upid) {
-      return baseColor.adjusted(baseColor.h, baseColor.s - 20, Math.min(baseColor.l + 20,  60));
+      return colorForThread(queried, StyleConstants.isLight()? 3 : -3);
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/perfetto/models/ThreadInfo.java]: nit: missing space

File: gapic/src/main/com/google/gapid/perfetto/models/ThreadInfo.java [hunk 3]
```diff
@@ -96,20 +98,30 @@ public String getDisplay() {
...
     } else {
-      return StyleConstants.getGrayColor();
+      return StyleConstants.getGrayColor().rgb();
+    }
+  }
+
+  public static RGBA getBorderColor(State state, long utid) {
+    return colorForThread(state.getThreadInfo(utid), StyleConstants.isLight()? -5 : 5);
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/perfetto/models/ThreadInfo.java]: nit: missing space

File: gapic/src/main/com/google/gapid/perfetto/views/StyleConstants.java [hunk 4]
```diff
@@ -274,52 +174,142 @@ public static Colors dark() {
...
-      new HSL(16, 25, 38),
-      new HSL(200, 18, 46),
-      new HSL(54, 100, 62),
-  };
   private static final HSL GRAY_COLOR = new HSL(0, 0, 62);
 
+  /*
+   * A color scheme based on "ADUX Dataviz Palette":
+   * https://docs.google.com/spreadsheets/d/12kGr47Y5kavpBatlu4Co61UgFh7OMY9_DMg9Sd0viHI.
+   * */
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/perfetto/views/StyleConstants.java]: Remove. This is an internal link that should not be part of the code.

File: gapic/src/main/com/google/gapid/perfetto/views/StyleConstants.java [hunk 5]
```diff
@@ -274,52 +174,142 @@ public static Colors dark() {
...
+        new HSL(262, 70, 58),
+        new HSL(22, 76, 43),
+        new HSL(217, 50, 40),
+        new HSL(59, 51, 45),
+        new HSL(320, 53, 47),
+        new HSL(215, 100, 55),
+    };
+
+    // Note: The color names correspond to HSL values in <BASE_COLORS>. Update simultaneously.
+    public static enum Name {
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/perfetto/views/StyleConstants.java]: Combine this with HSL into a single enum

File: gapic/src/main/com/google/gapid/perfetto/views/StyleConstants.java [hunk 6]
```diff
@@ -274,52 +174,142 @@ public static Colors dark() {
...
+      LIME,
+      MAGENTA,
+      VIVID_BLUE,
+    }
+
+    private static final int COLOR_COUNT = BASE_COLORS.length;
+    private static final int[] LIGHTULA_OFFSETS = new int[] {
+        5, 2, 4, 5, 4, 5, 6, 3, 10, 12, 6, 9, 10, 9, 9, 7,
+    };
+    private static final int[] DARCULA_OFFSETS = new int[] {
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/perfetto/views/StyleConstants.java]: Don't use the terms "lightula" and "darcula" they are IJ/AS terms. Simply "light" and "dark" for us.

File: gapic/src/main/com/google/gapid/perfetto/models/SliceTrack.java (hunk 1 above)
Comment (stellama0208) [reply gapic/src/main/com/google/gapid/perfetto/models/SliceTrack.java]: Interesting to learn, thanks!

**Comment by stellama0208** [comment]
The above comments are addressed, ready for another look, thanks!

**Comment by stellama0208** [comment]
The above force push is only to squash commits and resolve merge conflicts.

**Pull request merged**